alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt

```diff
@@ -867,4 +868,228 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
+        // Create monitor #1
+        var monitorResponse = createMonitor(monitor)
+        assertFalse(monitorResponse?.id.isNullOrEmpty())
+        monitor = monitorResponse!!.monitor
+        // Execute monitor #1
+        var executeMonitorResponse = executeMonitor(monitor, monitorResponse.id, false)
+        Assert.assertEquals(executeMonitorResponse!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse.monitorRunResult.triggerResults.size, 1)
+        // Create monitor #2
+        var monitorResponse2 = createMonitor(monitor)
+        assertFalse(monitorResponse2?.id.isNullOrEmpty())
+        monitor = monitorResponse2!!.monitor
+        // Insert doc #2. This one should trigger creation of alerts during monitor exec
+        val testDoc = """{
+            "test_field" : "us-west-2"
+        }"""
+        indexDoc(testSourceIndex, "2", testDoc)
+        // Execute monitor #2
+        var executeMonitorResponse2 = executeMonitor(monitor, monitorResponse2.id, false)
+        Assert.assertEquals(executeMonitorResponse2!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse2.monitorRunResult.triggerResults.size, 1)
+
+        refreshIndex(AlertIndices.ALERT_INDEX)
+        var alerts = searchAlerts(monitorResponse2.id)
+        Assert.assertTrue(alerts != null)
+        Assert.assertTrue(alerts.size == 1)
+
+        // Both monitors used same queryIndex. Since source index has close to limit amount of fields in mappings,
+        // we expect that creation of second monitor would rollover queryIndex
+        val getIndexResponse: GetIndexResponse =
+            client().admin().indices().getIndex(GetIndexRequest().indices(ScheduledJob.DOC_LEVEL_QUERIES_INDEX + "*")).get()
+        assertEquals(1, getIndexResponse.indices.size)
+        // Now we'll verify that execution of both monitors work
+        indexDoc(testSourceIndex, "3", testDoc)
+        // Exec Monitor #1
+        executeMonitorResponse = executeMonitor(monitor, monitorResponse.id, false)
+        Assert.assertEquals(executeMonitorResponse!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse.monitorRunResult.triggerResults.size, 1)
+        refreshIndex(AlertIndices.ALERT_INDEX)
+        alerts = searchAlerts(monitorResponse.id)
+        Assert.assertTrue(alerts != null)
+        Assert.assertTrue(alerts.size == 2)
+        // Exec Monitor #2
+        executeMonitorResponse = executeMonitor(monitor, monitorResponse2.id, false)
+        Assert.assertEquals(executeMonitorResponse!!.monitorRunResult.monitorName, monitor.name)
+        Assert.assertEquals(executeMonitorResponse.monitorRunResult.triggerResults.size, 1)
+        refreshIndex(AlertIndices.ALERT_INDEX)
+        alerts = searchAlerts(monitorResponse2.id)
+        Assert.assertTrue(alerts != null)
+        Assert.assertTrue(alerts.size == 2)
+    }
+
+    fun `test queryIndex bwc`() {
```

Comment:
Update name to `test queryIndex bwc when index was not an alias`

Reply:
done!